spec/tpm/key_attestation_spec.rb

```diff
@@ -167,4 +167,39 @@
       end
     end
   end
+
+  describe "#key" do
+    context "when everything's in place" do
+      it "returns an RSA key" do
+        expect(key_attestation.key).to be_a OpenSSL::PKey::RSA
+      end
+    end
+
+    context "when RSA PSS algorithm" do
+      before do
+        unless OpenSSL::PKey::RSA.instance_methods.include?(:verify_pss)
+          skip "Ruby OpenSSL gem #{OpenSSL::VERSION} do not support RSASSA-PSS"
+        end
+      end
+
+      let(:signature_algorithm) { TPM::ALG_RSAPSS }
+      let(:signature) do
+        attestation_key.sign_pss(hash_function, to_be_signed, salt_length: :max, mgf1_hash: hash_function)
+      end
+
+      it "returns an RSA key" do
+        expect(key_attestation.key).to be_a OpenSSL::PKey::RSA
+      end
+    end
+
+    context "when is not valid" do
+      before do
+        expect(key_attestation).to receive(:valid?).and_return(false)
+      end
+
+      it 'returns nil' do
+        expect(key_attestation.key).to be nil
+      end
+    end
```

Comment:
I wanted to avoid re-writing all the cases in which `KeyAttestation` could be invalid, since we doing that above, in order to avoid duplicated code

Reply:
Sounds good :+1: